MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -2729,7 +2729,7 @@ public boolean canPayFor(IAcquisitionWork acquisition) {
      * @return true if your target roll succeeded.
      */
     public boolean findContactForAcquisition(IAcquisitionWork acquisition, Person person, PlanetarySystem system) {
-        if (person.getAcquisitions() >= getCampaignOptions().getMaxAcquisitions()) {
+        if (null == person || !canAcquireParts(person)) {
```

Comment:
This was causing the method to return false when the user left max acquisitions at zero which is supposed to be unlimited. Using the `canAcquireParts` method instead fixes this because it already checks that properly.

Reply:
This is already checked in goShoppingByPlanet(ShoppingList), which is the only place this method is called. A null person is also not possible from the current path.

Reply:
Good point. I have removed this entirely now as its redundant.